### ISSUE TITLE
Windows with enforced DEP doesn't support emulation with dynarac

### ISSUE THREAD

**Issue by richard42** [mupen64plus/mupen64plus-user-issues#360]
Originally reported on Google Code with ID 361

```
rehack reported via irc that he wanted to run his emulator in windows xp
64bit sp2 with enforced DEP (NX memory protection) for all programs. The
emulator just stops without further notice.

He whitelisted m64p and it just ran afterwards.

Solution is to write malloc_exec for windows. An example was given by him:

void *malloc_exec(size_t size)
{
 return VirtualAlloc(NULL, size, MEM_COMMIT | MEM_RESERVE,
PAGE_EXECUTE_READWRITE);
}

He also said that a free function must be implemented (but maybe also
works with free...)

VirtualFree(address, 0, MEM_RELEASE)

```

Reported by `sven@narfation.org` on 2010-04-04 19:14:35

**Comment by richard42** [comment]
```
free(...) to change to free_exec are in

 * realloc_exec
 * interupt.c: if (blocks[i]->block) { free(blocks[i]->block); blocks[i]->block =
NULL; }
 * r4300.c: if (blocks[i]->block) { free(blocks[i]->block); blocks[i]->block = NULL;
}

I don't have something to test it - so I just assume that this is correct.
```

Reported by `sven@narfation.org` on 2010-04-04 19:24:24

**Comment by richard42** [comment]
```
I think it should be fixed by 6bb44b4ffdd4
```

Reported by `sven@narfation.org` on 2010-04-06 20:03:37

**Comment by richard42** [comment]
Reported by `richard42g` on 2010-04-10 01:36:28
- **Status changed**: `Fixed`
